doc/adr/adr-003.md

```diff
@@ -0,0 +1,19 @@
+### ADR - 003: Form Submission Flow
+
+Turbo Frames (https://github.com/hotwired/turbo-rails) for form pages/questions with data saved (but not necessarily fully validated) to Active Record model on each submit.
+
+
+#### Impact on Performance
+
+Using Turbo Frames allows us to swap out just the question part of the page without needing full page refreshes as you go through the form and provides an "SPA like" user experience. Each question still gets a unique URL that can be navigated to directly with the Case Log ID and the overall user experience is that form navigation feels faster.
```

Comment:
Could be worth using Single Page Application instead of the acronym SPA just for clarity here?